java/code/src/com/suse/manager/reactor/messaging/test/SaltTestUtils.java

```diff
@@ -167,6 +176,56 @@ public static <T> Matcher<Call<T>> functionEquals(String module, String func) {
                 Matchers.hasProperty("functionName", equalTo(func)));
     }
 
+    /**
+     * Simple in-memory appender to capture log events on a given logger
+     */
+    public static class TestLogAppender extends AbstractAppender {
+
+        private final List<LogEvent> log;
+
+        public TestLogAppender(String name, List<LogEvent> testLog) {
+            super(name, null, null);
+            this.log = testLog;
+        }
+
+        public boolean matchInLogs(String regex) {
+            var pattern = Pattern.compile(regex);
+            return log
+                    .stream()
+                    .anyMatch(e -> pattern.matcher(e.getMessage().getFormattedMessage()).find());
+        }
+
+        public List<LogEvent> getLog() {
+            return log;
+        }
+
+        @Override
+        public void append(LogEvent logEvent) {
+            log.add(logEvent);
+        }
+
+    }
+
+    /**
+     * Injects a simple in-memory appender to capture log events on a given logger
+     * @param cls the class to associate the new appender with
+     * @return the in-memory logs appender
+     */
+    public static TestLogAppender enableTestLogging(Class<?> cls) {
+        TestLogAppender listAppender = new TestLogAppender("testAppender", new ArrayList<>());
+        listAppender.start();
+
+        LoggerContext loggerContext = (LoggerContext) LogManager.getContext(cls.getClassLoader(), false);
+        Configuration configuration = loggerContext.getConfiguration();
+        configuration.addAppender(listAppender);
+
+        LoggerConfig rootLoggerConfig = configuration.getLoggerConfig("");
+        rootLoggerConfig.removeAppender("rootAppender");
+        rootLoggerConfig.addAppender(listAppender, Level.ERROR, null);
+
+        return listAppender;
+    }
+
```

Comment:
@cbosdo capturing the logs doesn't seem to be an easy task with log4j2, at least it's not well documented. confused But I guess I've found a solution which is easy enough to be used in other cases too.

Reply:
Seems a nice solution to me 👍🏻